lib/internal/per_context/primordials.js

```diff
@@ -84,21 +84,31 @@ primordials.makeSafe = makeSafe;
 
 // Subclass the constructors because we need to use their prototype
 // methods later.
+// Defining the `constructor` is necessary here to avoid the default
+// constructor which uses the user-mutable `%ArrayIteratorPrototype%.next`.
 primordials.SafeMap = makeSafe(
   Map,
-  class SafeMap extends Map {}
+  class SafeMap extends Map {
+    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
```

Comment:
```suggestion
    constructor(iterable = undefined) {
      super(iterable);
    }
```

Reply:
You're talking about `SafeMap.prototype.constructor.length`, right? That doesn't seem necessary, and I prefer to keep the eslint comment to make it obvious to future reader that this was done on purpose. Unless there is a context where `length` value matters in the code base?